src/App.tsx

```diff
@@ -1,61 +1,135 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+const getUser = (id: number): User | null => {
+  return usersFromServer.find(user => user.id === id) || null;
+};
+
+const todoList: Todo[] = todosFromServer.map(todo => (
+  {
+    ...todo,
+    user: getUser(todo.userId),
+  }
+));
+
+const getTodoId = (todos: Todo[]): number => {
+  return Math.max(...todos.map(todo => todo.id)) + 1;
+};
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(todoList);
+  const [userError, showUserError] = useState(false);
+  const [titleError, showTitleError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    showTitleError(false);
+  };
+
+  const handleUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(Number(event.target.value));
+    showUserError(false);
+  };
+
+  const clear = () => {
+    showUserError(false);
+    showTitleError(false);
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
```

Comment:
Consider using `event.stopPropagation()` to prevent the form from submitting if there are errors. 🛑